docs/source/l1_to_l2b_e2e.rst

```diff
@@ -0,0 +1,220 @@
+    this_caldb = caldb.CalDB()
+       
+
+Create necessary calibration products including nonlinearity and KGain
+
+.. code-block:: python
+
+    # Nonlinearity
+    nonlin_path = os.path.join(processed_cal_path, "nonlin_table_240322.txt")
+    nonlin_dat = np.genfromtxt(nonlin_path, delimiter=",")
+    nonlinear_cal = data.NonLinearityCalibration(nonlin_dat, pri_hdr, ext_hdr, mock_input_dataset)
+    nonlinear_cal.save(test_outputdir, "mock_nonlinearcal.fits")
+    this_caldb.create_entry(nonlinear_cal)
+
+    # KGain
+    kgain = data.KGain(np.array([[8.7]]), pri_hdr, ext_hdr, mock_input_dataset)
+    kgain.save(test_outputdir, "mock_kgain.fits")
+    this_caldb.create_entry(kgain)
+
+Calibration files such as non-linearity tables, dark current, flat fields, and bad pixel maps are loaded.
+
+.. code-block:: python
+
+    def load_fits(path): return fits.open(path)[0].data
+
+    fpn = load_fits(os.path.join(processed_cal_path, "fpn_20240322.fits"))
+    cic = load_fits(os.path.join(processed_cal_path, "cic_20240322.fits"))
+    dark = load_fits(os.path.join(processed_cal_path, "dark_current_20240322.fits"))
+
+    noise_map_dat_img = np.array([fpn, cic, dark])
+    noise_map_dat = np.zeros((3, detector.detector_areas['SCI']['frame_rows'], detector.detector_areas['SCI']['frame_cols']))
+    rows, cols, r0c0 = detector.unpack_geom('SCI', 'image')
+    noise_map_dat[:, r0c0[0]:r0c0[0]+rows, r0c0[1]:r0c0[1]+cols] = noise_map_dat_img
+
+    noise_map = data.DetectorNoiseMaps(noise_map_dat, pri_hdr, ext_hdr, mock_input_dataset,
+                                       err=np.zeros([1] + list(noise_map_dat.shape)),
+                                       dq=np.zeros(noise_map_dat.shape, dtype=int),
+                                       err_hdr=fits.Header({'BUNIT': 'detected electrons'}))
+    noise_map.save(test_outputdir, "mock_detnoisemaps.fits")
+    this_caldb.create_entry(noise_map)
+
+    # Flat and bad pixel map
+    flat = data.FlatField(load_fits(os.path.join(processed_cal_path, "flat.fits")), pri_hdr, ext_hdr, mock_input_dataset)
+    flat.save(test_outputdir, "mock_flat.fits")
+    this_caldb.create_entry(flat)
+
+    bp = data.BadPixelMap(load_fits(os.path.join(processed_cal_path, "bad_pix.fits")), pri_hdr, ext_hdr, mock_input_dataset)
+    bp.save(test_outputdir, "mock_bpmap.fits")
+    this_caldb.create_entry(bp)
+
+Piepline Execution
+~~~~~~~~~~~~~~~~~~~
+
+Execute the pipeline to process L1 data through L2b:
+The ``walker.walk_corgidrp`` function is the main part of the pipeline responsible for transforming the raw L1 data into L2b data. This function applies all necessary calibration steps and generates the output files.
+
+.. code-block:: python
+
+    walker.walk_corgidrp(l1_data_filelist, "", l2a_outputdir)
+    new_l2a_filenames = [os.path.join(l2a_outputdir, f"{i}.fits") for i in [90499, 90500]]
+    walker.walk_corgidrp(new_l2a_filenames, "", l2b_outputdir)
+
+Validate the pipeline results:
```

Comment:
I don't think we need this verification of pipeline results. The focus on this tutorial is just to show how the pipeline works, so I don't think we need this.